internal/subscription/core.go

```diff
@@ -344,6 +348,106 @@ func createDelayTopics(ctx context.Context, m *Model, topicCore topic.ICore, top
 	return nil
 }
 
+// RescaleSubTopics - Get all the subs and rescale all the Retry/Delay/DLQ topics
+func (c *Core) RescaleSubTopics(ctx context.Context, topicModel *topic.Model) error {
+	projectList, err := c.projectCore.ListKeys(ctx)
+	if err != nil {
+		return err
+	}
+	var completeSubList []*Model
+
+	for _, projectKey := range projectList {
+		subList, subErr := c.List(ctx, Prefix+project.FetchProjectID(projectKey))
+		if subErr != nil {
+			return subErr
+		}
+		completeSubList = append(completeSubList, subList...)
+	}
+	admin, err := c.brokerStore.GetAdmin(ctx, messagebroker.AdminClientOptions{})
+	if err != nil {
+		return merror.ToGRPCError(err)
+	}
+
+	for _, m := range completeSubList {
+		if m.ExtractedTopicName != topicModel.ExtractedTopicName {
```

Comment:
The comparison should be between `sub.Topic` and `topicModel.Name`. Extracted names ignore project for the topic